.github/workflows/pull-request.yml

```diff
@@ -49,8 +49,8 @@ jobs:
       - name: Install Dependencies
         run: npm ci
 
-      - name: Run ESLint to check for linting errors
-        run: npm run lint:check
+      - name: Run ESLint to check for linting errors in modified files
+        run: npx eslint $(git diff --name-only --diff-filter=d ${{ github.event.before }} ${{ github.sha }})
```

Comment:
- No need to use `npx`
- This way is not secure as the the git diff may include files of extensions that we don't want to lint. What happened to the `lint-staged` based approach?

Reply:
1. When you say no need to use npx, do you mean to exclude it from the line or rather use `npm run`?
2.`npx eslint $(git diff --name-only --diff-filter=d -- '***.ts' ${{ github.event.before }} ${{ github.sha }})` Using `'***.ts'` with the git diff we can limit the extensions.

Reply:
1. Use `npm run`.
2. There is no need to make these changes here. Linting and type-checking should be performed on the whole base in PR workflows.

Reply:
@EshaanAgg 

1. The issue is that there are so many code base linting errors that all our PRs will fail and no one will want to lint someone else's code especially if that code was created years ago.
2. We need to lint deltas for both PRs and commits to the local branch so that we can incrementally improve the code base.
3. When this PR is completed and we know it works, then we can create issues to lint the entire code base.

@EshaanAgg @NamitBhutani Let's proceed with the delta approach as stated in the original issue

Reply:
@NamitBhutani Add the filter for only `.tsx` files

Reply:
> @NamitBhutani Add the filter for only `.tsx` files

Do you mean in `npx eslint $(git diff --name-only --diff-filter=d -- '***.ts' ${{ github.event.before }} ${{ github.sha }})`. Also the repo has no .tsx files.

Reply:
Yes, my error, Typescript files